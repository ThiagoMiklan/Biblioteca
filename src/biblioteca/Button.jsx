// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import {renderToString} from 'react-dom/server';

type Props = {
  disabled?: bool,
  delete?: bool,
  definition?: string,
  children?: React.Node,
  custom?: any,
  onClick?: () => void
}

/*

Possui children
Eventos: onClick no <button> </button>
Observações: 

    1 - Caso exista a necessidade de adicionar um ícone ao componente Button, é necessário seguir
             a seguinte estrutura:

            <Button definition="info" >
                <Icon icon_name="fas fa-check" [right={true} OR empty prop right]>
                    Nome do botão
                </Icon>
            </Button >

            Sendo necessário informar o componente Icon como filho do componente Button
            e o nome do botão deve ser informado no Icon e não no Button como em outras
            situações onde usa-se o componente isolado.
*/

const Button= (props: Props)=>{
    var definition = getDefinition(props);
    
     return <button 
                {...props.custom} 
                disabled = {props.disabled == true? true: false } 
                className={definition} 
                onClick={props.onClick}
                >
                {props.children}
         </button>
         ;
}

function getDefinition(props){
    var definition = "";

    if(props.delete ==  true){
        definition = "delete";
    }else{
        definition = (props.definition == undefined) ? "button" : "button "+ props.definition;
    }

    return definition;
}

Button.propTypes = {
    definition: PropTypes.string,
    onClick: PropTypes.func,
    delete: PropTypes.bool,
    disabled: PropTypes.bool
};


export default Button;