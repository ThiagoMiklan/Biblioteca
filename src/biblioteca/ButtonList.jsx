// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassNames from '../tools/getClassName';


// Componente criado com o objetivo de servir de containe para diversos botões
// Oferece possibiidade de escolher se o container é "field" ou "buttons", para escolher field informar field={true}
// Class Field : Funcona com diversos elementos e não somente botões
// Class Buttons: Funciona apenas com botões

type Props ={
    definition?: string,
    field?: bool,
    children?:React.Node
}

const ButtonList = (props:Props) => {
    return (assembleButtonList(props));
}

function assembleButtonList(props) {
    var code_icon = "";
    var definition = getDefinition(props.definition, props.field);

    // permite o funcionamento dos addons, pois fornece uma interação que gera os filhos dentro de uma tag <p className=control>
    if (definition.includes("addons")) {
        code_icon =
            <div className={definition}>
                {
                    React.Children.map(props.children, (child, i) => {
                        return <p className="control">
                            <div className={definition}>
                                {child}
                            </div>
                        </p>
                    })
                }
             </div>

    } else {
        code_icon =
            <div className={definition}>
                {props.children}
            </div>;
    }

    return code_icon;
}

function getDefinition(definition:any, field) {

    if (field == true) {
        definition = classnames(getClassNames("field " + definition, "ButtonList"));;
    } else {
        definition = classnames(getClassNames("buttons " + definition, "ButtonList"));;
    }
    return definition;
}

ButtonList.propTypes = {
    definition: PropTypes.string,
    field: PropTypes.bool
};

export default ButtonList;