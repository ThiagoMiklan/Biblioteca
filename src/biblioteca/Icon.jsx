// @flow
import * as React from 'react'
import classnames from 'classnames';
import getClassNames from '../tools/getClassName';
import PropTypes from 'prop-types';
import '../css/all';
import checkerDefinition from '../tools/checker.js';



type Props = {
    definition?: string,
    icon_right?: bool,
    onClick? : ()=> void,
    children?: React.Node,
    icon_name?: string
}

const Icon = (props:Props) => {
    checkerDefinition(props.definition,"Icon");
    var icon_definition  = (props.definition == undefined) ? "": props.definition;
    var definition =  classnames(getClassNames("icon "+icon_definition, "Icon"));
    return (
        <>  
            {(props.icon_right == false || props.icon_right == undefined) && <span>{props.children}</span>}
            <span onClick={props.onClick} className={definition}>
                <i className={props.icon_name} aria-hidden="true" ></i>
            </span>
            {props.icon_right == true && <span>{props.children}</span>}
            
        </>
    );
}

function Teste(){
    var x = <Icon ic></Icon>
}




export default Icon;