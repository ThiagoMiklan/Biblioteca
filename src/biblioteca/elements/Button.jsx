import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import button_props from '../../props/ButtonProps.js'
import getClassNames from '../../tools/getClassName';

/*
Observações: 

    1 - Caso exista a necessidade de adicionar um ícone ao componente Button, é necessário seguir
             a seguinte estrutura:

            <Button definition="info" >
                <Icon icon_name="fas fa-check" [right={true} OR empty prop right]>
                    Nome do botão
                </Icon>
            </Button >

            Sendo necessário informar o componente Icon como filho do componente Button
            e o nome do botão deve ser informado no Icon e não no Button como em outras
            situações onde usa-se o componente isolado.
*/


class Button extends React.Component {

    constructor(props){
        super(props);
        this.state  = {definition:props.definition};
    }

    render() {
        var definition = getDefinition(this.props.delete, this.state.definition);

        return (
            <button disabled = {this.props.disabled == true? true: false } className={definition} onClick={getClickEvent(this.props.onClick).bind(this)}>
                {this.props.children}
         </button>
         );
    }
}

/* Se propriedade isDelete é verdadeira, então
    retorna somente o className "definition", não
    envolvendo outras classes do Bulma.
*/
function getDefinition(isDelete,definition){
    if(isDelete){
       return "delete";
    }else{
        return classnames(getClassNames("button "+ definition,"Button"));
    }
}
// caso seja informado onClick ele retorna a função retornada, senão retorna uma função vazia. Método criado para evitar problemas com undefined
function getClickEvent(onClick){

    if(typeof onClick == "function"){
        return onClick;
    }else{
        return ()=> {};
    }
    
}



Button.propTypes = {
    // definição CSS Bulma
    definition: PropTypes.string,
    // Método Onclick disponível
    onClick: PropTypes.func,
    // Caso exista um ícone, informar através da prop icon, passando o componente Icon
    icon: PropTypes.element,
    // Caso o botão seja do tipo delete
    delete: PropTypes.bool
};


export default Button;