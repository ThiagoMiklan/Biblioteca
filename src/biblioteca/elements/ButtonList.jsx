import React from 'react';
import classnames from 'classnames';
import getClassNames from '../../tools/getClassName';

const ButtonList = (props) => {
    var definition = classnames(getClassNames("buttons " + props.definition,"ButtonList"));
    return (<div className={definition}>
        {props.children}
    </div>
    );
}

export default ButtonList;