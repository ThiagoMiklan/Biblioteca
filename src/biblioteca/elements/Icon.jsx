import React from 'react'

// ProvisÃ³rio
const Icon = (props) => {
    return (
        <span class={props.definition}>
            <i class={props.icon}></i>
        </span>
    );
}

export default Icon;