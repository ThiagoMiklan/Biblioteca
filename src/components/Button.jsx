// @flow
import * as React from 'react';
import { validate } from '../tools/type_validations.js';

// custom

type Props = {
    disabled?: bool,
    delete?: bool,
    label?: React.Node,
    definition?: string,
    children?: React.Node,
    custom?: Object,
    onClick?: () => void
}

const props_obj = {
    disabled: 'boolean',
    delete: 'boolean',
    definition: 'string',
    custom: 'object',
    onClick: 'function'
}

const Button = (props: Props) => {
    validate(props, props_obj, "Button");
    var definition = getDefinition(props);
    return <button
        {...props.custom}
        disabled={props.disabled === true ? true : false}
        className={definition}
        onClick={props.onClick}
    >
        {props.label}
    </button>
        ;
}

function getDefinition(props) {
    var definition = "";

    if (props.delete === true) {
        definition = "delete";
    } else {
        definition = (props.definition === undefined) ? "button" : "button " + props.definition;
    }

    return definition;
}



export default Button;