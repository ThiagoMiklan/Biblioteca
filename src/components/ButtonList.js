// @flow
import * as React from 'react';
import {validate} from '../tools/type_validations.js';

type Props ={
    definition?: string,
    children?:React.Node,
    custom?: Object 
}

const props_obj ={
    definition: "string",
    custom: 'object'
}

const ButtonList = (props:Props) => {
    validate(props,props_obj,"ButtonList")
return assembleButtonList(props);
}

function assembleButtonList(props) {
    validate(props,props_obj,"ButtonList");
    var code_icon = "";
    var definition = (props.definition ==  undefined)? "buttons": "buttons "+props.definition;
    
        if (definition.includes("addons")) {
        code_icon =
            <div className={props.definition} {...props.custom}>
                {
                    React.Children.map(props.children, (child, i) => {
                        return <p className="control">
                            <div className={definition}>
                                {child}
                            </div>
                        </p>
                    })
                }
             </div>

    } else {
        code_icon =
            <div className={definition}>
                {props.children}
            </div>;
    }
  
    return code_icon;
}

export default ButtonList;