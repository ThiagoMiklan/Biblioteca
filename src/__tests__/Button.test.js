import React from 'react';
import Button from '../biblioteca/Button';
import ButtonList from '../biblioteca/ButtonList';
import Icon from '../biblioteca/Icon';
import renderer from 'react-test-renderer';


function onClick() {
  alert(testando);
}

test('value correctly', () => {
  const componentWithValue = renderer
    .create(<Button value="Facebook" />)
    .toJSON();
  // componente com valor
  expect(componentWithValue).toMatchSnapshot();

  // componente sem valor
  const componentWithoutValue = renderer.create(<Button />)
    .toJSON();
  expect(componentWithoutValue).toMatchSnapshot();

});

// testa se as cores estão corretas, cores simples sem características adicionais
test('color correctly', () => {

  const simple_color = renderer.create(<ButtonList>
    <Button>Testando</Button>
    <Button  definition="primary">Testando</Button>
    <Button  definition="link"> Testando</Button>
    <Button  definition="info"> Testando</Button>
    <Button  definition="success"> Testando</Button>
    <Button  definition="warning"> Testando</Button>
    <Button  definition="danger"> Testando</Button>
    </ButtonList>
    ).toJSON();

  expect(simple_color).toMatchSnapshot();
});

// testa se as cores estão corretas, cores secundárias dos botões
test('color secundary correctly', () => {

  const secundary_color = renderer.create(
    <ButtonList>
    <Button  definition="white"> Testando</Button>
    <Button  definition="light"> Testando</Button>
    <Button  definition="dark"> Testando</Button>
    <Button  definition="black"> Testando</Button>
    <Button  definition="text"> Testando</Button>
    </ButtonList>).toJSON();

  expect(secundary_color).toMatchSnapshot();
});

// testa se as cores estão corretas, cores com característica claras
test('colors light correctly', () => {

  const light_color = renderer.create(<ButtonList>
    <Button  definition="primary light"> Testando</Button>
    <Button  definition="link light"> Testando</Button>
    <Button  definition="info light"> Testando</Button>
    <Button  definition="success light"> Testando</Button>
    <Button  definition="warning light"> Testando</Button>
    <Button  definition="danger light"> Testando</Button>
    
    </ButtonList>).toJSON();

  expect(light_color).toMatchSnapshot();
});

// testa se os tamanhos dos botões estão corretos
test('buttons size correctly', () => {

  const light_color = renderer.create(<ButtonList>
    <Button  definition="button small"> Testando</Button>
    <Button  > Testando</Button>
    <Button  definition="button normal"> Testando</Button>
    <Button  definition="button medium"> Testando</Button>
    <Button  definition="button large"> Testando</Button>
    </ButtonList>
    ).toJSON();

  expect(light_color).toMatchSnapshot();
});

// testa se os tamanhos dos botões estão corretos
test('button list size correctly', () => {

  const small = renderer.create(
    <ButtonList definition="small">
<Button  > Testando</Button>
<Button  > Testando</Button>
<Button  > Testando</Button>
</ButtonList>).toJSON();
  expect(small).toMatchSnapshot();

  const medium = renderer.create(
    <ButtonList definition="medium">
<Button > Testando</Button>
<Button > Testando</Button>
<Button > Testando</Button>
</ButtonList>).toJSON();
  expect(medium).toMatchSnapshot();

  const large = renderer.create(
    <ButtonList definition="large">
<Button > Testando</Button>
<Button > Testando</Button>
<Button > Testando</Button>
</ButtonList>).toJSON();
  expect(large).toMatchSnapshot();
});

// testa se a característica fullwidth está correta
test('fullwidth  correctly', () => {

  const fullwidth = renderer.create(<ButtonList>
    <Button  definition="small fullwidth"> Testando</Button>
    <Button  definition="fullwidth"> Testando</Button>
    <Button  definition="medium fullwidth"> Testando</Button>
    <Button definition="large fullwidth"> Testando</Button>
    </ButtonList>
    ).toJSON();

  expect(fullwidth).toMatchSnapshot();
});

// testa se a característica outlined está correta
test('outlined  correctly', () => {
  const outlined = renderer.create(<ButtonList>
    <Button  definition="primary outlined"> Testando</Button>
    <Button  definition="link outlined"> Testando</Button>
    <Button  definition="info outlined"> Testando</Button>
    <Button  definition="success outlined"> Testando</Button>
    <Button  definition="warning outlined"> Testando</Button>
    <Button  definition="danger outlined"> Testando</Button>
    </ButtonList>
    ).toJSON();

  expect(outlined).toMatchSnapshot();
});

// testa se a característica inverted está correta
test('inverted  correctly', () => {
  const inverted = renderer.create(
    <ButtonList definition="primary">
<Button  definition="primary inverted"> Testando</Button>
<Button  definition="link inverted"> Testando</Button>
<Button  definition="info inverted"> Testando</Button>
<Button  definition="success inverted"> Testando</Button>
<Button  definition="danger inverted"> Testando</Button>
</ButtonList>
  ).toJSON();

  expect(inverted).toMatchSnapshot();
});

// testa se a característica rounded está correta
test('rounded  correctly', () => {
  const rounded = renderer.create(
    <ButtonList>
    <Button  definition="primary rounded"> Testando</Button>
    <Button  definition="link rounded"> Testando</Button>
    <Button  definition="info rounded"> Testando</Button>
    <Button  definition="success rounded"> Testando</Button>
    <Button  definition="warning rounded"> Testando</Button>
    <Button  definition="danger rounded"> Testando</Button>
    </ButtonList>
    

  ).toJSON();

  expect(rounded).toMatchSnapshot();
});

// testa os estados
test('state  correctly', () => {

  const hovered = renderer.create(<ButtonList>
    <Button  > Testando</Button>
    <Button  definition="primary hovered"> Testando</Button>
    <Button  definition="link hovered"> Testando</Button>
    <Button  definition="info hovered"> Testando</Button>
    <Button  definition="success hovered"> Testando</Button>
    <Button  definition="warning hovered"> Testando</Button>
    <Button  definition="danger hovered"> Testando</Button>
    </ButtonList>
    
    ).toJSON();
  expect(hovered).toMatchSnapshot();

  const focused = renderer.create(<ButtonList>
    <Button  />
    <Button  definition="primary focused"> Testando</Button>
    <Button  definition="link focused"> Testando</Button>
    <Button  definition="info focused"> Testando</Button>
    <Button  definition="success focused"> Testando</Button>
    <Button  definition="warning focused"> Testando</Button>
    <Button  definition="danger focused"> Testando</Button>
    </ButtonList>).toJSON();

  expect(focused).toMatchSnapshot();

  const active = renderer.create(
    <ButtonList>
    <Button  />
    <Button  definition="primary active"> Testando</Button>
    <Button  definition="link active"> Testando</Button>
    <Button  definition="info active"> Testando</Button>
    <Button  definition="success active"> Testando</Button>
    <Button  definition="warning active"> Testando</Button>
    <Button  definition="danger active"> Testando</Button>
    </ButtonList>
    
    ).toJSON();

  expect(active).toMatchSnapshot();

});

test('loading corretly', () => {
  const loading = renderer.create(<ButtonList>
    <Button  definition="loading"> Testando</Button>
    <Button  definition="primary loading"> Testando</Button>
    <Button  definition="link loading"> Testando</Button>
    <Button  definition="info loading"> Testando</Button>
    <Button  definition="success loading"> Testando</Button>
    <Button  definition="warning loading"> Testando</Button>
    <Button  definition="danger loading"> Testando</Button>
    </ButtonList>).toJSON();

  expect(loading).toMatchSnapshot();

});

test('disabled corretly', () => {
  const disabled = renderer.create(<ButtonList>
    <Button  disabled={true} > Testando</Button>
    <Button  definition="primary" disabled={true} > Testando</Button>
    <Button  definition="link"  disabled={true} > Testando</Button>
    <Button  definition="info" disabled={true} > Testando</Button>
    <Button  definition="success" disabled={true} > Testando</Button>
    <Button  definition="warning" disabled={true} > Testando</Button>
    <Button  definition="danger" disabled={true} > Testando</Button>
    </ButtonList>).toJSON();

  expect(disabled).toMatchSnapshot();

});

test('buttons with icons',()=>{
  const buttons_icons = renderer.create(
    <ButtonList>
  <Button>
  <Icon icon_name="fas fa-bold"></Icon>
  </Button>
  
  <Button>
  <Icon icon_name="fas fa-italic"></Icon>
  </Button>

  <Button>
  <Icon icon_name="fas fa-underline"></Icon>
  </Button>

  <Button>
  <Icon icon_name="fab fa-github">
    Github
  </Icon>
  </Button>
  
  <Button definition="primary">
  <Icon icon_name="fab fa-twitter">
    Twitter
  </Icon>
  </Button>
  
  <Button definition="success">
  <Icon icon_name="fas fa-check">
    Save
  </Icon>
  </Button>

  <Button definition="danger outlined">
  <Icon icon_name="fas fa-times" icon_right={true}>
    Delete
  </Icon>
  </Button>

  <Button definition="small">
  <Icon icon_name="fab fa-github">
    Github
  </Icon>
  </Button>

  <Button >
  <Icon icon_name="fab fa-github">
    Github
  </Icon>
  </Button>

  <Button definition="medium">
  <Icon icon_name="fab fa-github">
    Github
  </Icon>
  </Button>

  <Button definition="large">
  <Icon icon_name="fab fa-github">
    Github
  </Icon>
  </Button>
</ButtonList>
  
 );

 expect(buttons_icons).toMatchSnapshot();
});

