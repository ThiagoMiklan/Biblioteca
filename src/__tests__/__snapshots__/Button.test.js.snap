// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button list size correctly 1`] = `
<div
  className="buttons are-small"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`button list size correctly 2`] = `
<div
  className="buttons are-medium"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`button list size correctly 3`] = `
<div
  className="buttons are-large"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`buttons size correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-small"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-normal"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-medium"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-large"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`buttons with icons 1`] = `
<div
  className="buttons"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
    <span />
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fas fa-bold"
      />
    </span>
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
    <span />
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fas fa-italic"
      />
    </span>
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
    <span />
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fas fa-underline"
      />
    </span>
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
    <span>
      Github
    </span>
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fab fa-github"
      />
    </span>
  </button>
  <button
    className="button is-primary"
    disabled={false}
    onClick={[Function]}
  >
    <span>
      Twitter
    </span>
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fab fa-twitter"
      />
    </span>
  </button>
  <button
    className="button is-success"
    disabled={false}
    onClick={[Function]}
  >
    <span>
      Save
    </span>
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fas fa-check"
      />
    </span>
  </button>
  <button
    className="button is-danger is-outlined"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fas fa-times"
      />
    </span>
    <span>
      Delete
    </span>
  </button>
  <button
    className="button is-small"
    disabled={false}
    onClick={[Function]}
  >
    <span>
      Github
    </span>
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fab fa-github"
      />
    </span>
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
    <span>
      Github
    </span>
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fab fa-github"
      />
    </span>
  </button>
  <button
    className="button is-medium"
    disabled={false}
    onClick={[Function]}
  >
    <span>
      Github
    </span>
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fab fa-github"
      />
    </span>
  </button>
  <button
    className="button is-large"
    disabled={false}
    onClick={[Function]}
  >
    <span>
      Github
    </span>
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fab fa-github"
      />
    </span>
  </button>
</div>
`;

exports[`color correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
    Testando
  </button>
  <button
    className="button is-primary"
    disabled={false}
    onClick={[Function]}
  >
    Testando
  </button>
  <button
    className="button is-link"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-info"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-success"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-warning"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-danger"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`color secundary correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-white"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-light"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-dark"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-black"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-text"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`colors light correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-primary is-light"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-link is-light"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-info is-light"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-success is-light"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-warning is-light"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-danger is-light"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`disabled corretly 1`] = `
<div
  className="buttons"
>
  <button
    className="button"
    disabled={true}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-primary"
    disabled={true}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-link"
    disabled={true}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-info"
    disabled={true}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-success"
    disabled={true}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-warning"
    disabled={true}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-danger"
    disabled={true}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`fullwidth  correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-small is-fullwidth"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-fullwidth"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-medium is-fullwidth"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-large is-fullwidth"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`inverted  correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-primary is-inverted"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-link is-inverted"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-info is-inverted"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-success is-inverted"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-danger is-inverted"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`loading corretly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-loading"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-primary is-loading"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-link is-loading"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-info is-loading"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-success is-loading"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-warning is-loading"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-danger is-loading"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`outlined  correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-primary is-outlined"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-link is-outlined"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-info is-outlined"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-success is-outlined"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-warning is-outlined"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-danger is-outlined"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`rounded  correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-primary is-rounded"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-link is-rounded"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-info is-rounded"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-success is-rounded"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-warning is-rounded"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-danger is-rounded"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`state  correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-primary is-hovered"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-link is-hovered"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-info is-hovered"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-success is-hovered"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-warning is-hovered"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-danger is-hovered"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`state  correctly 2`] = `
<div
  className="buttons"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="button is-primary is-focused"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-link is-focused"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-info is-focused"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-success is-focused"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-warning is-focused"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-danger is-focused"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`state  correctly 3`] = `
<div
  className="buttons"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
  />
  <button
    className="button is-primary is-active"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-link is-active"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-info is-active"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-success is-active"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-warning is-active"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
  <button
    className="button is-danger is-active"
    disabled={false}
    onClick={[Function]}
  >
     Testando
  </button>
</div>
`;

exports[`value correctly 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
/>
`;

exports[`value correctly 2`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
/>
`;
